src/main/java/co/cask/http/PatternPathRouterWithGroups.java

```diff
@@ -91,8 +91,10 @@ public void add(final String source, final T destination) {
       sb.append("/");
     }
 
-    //Ignore the last "/"
-    sb.setLength(sb.length() - 1);
+    // Ignore the last "/"
+    if (sb.length() > 0) {
```

Comment:
`patternRouteList` is a map of `path pattern -> the class to be routed to`. 

For `@Path("/")`, we expect an entry `"/" -> emptyPathClass`, however the split in line 73 does not return trailing empty strings, hence the entry becomes `"" -> emptyPathClass`. It would be good to keep `/` as path in `patternRouteList`. To do this we need to change the split to return empty trailing strings, which can be done by passing in the limit for split -
```
  String [] parts = path.split("/", maxPathParts + 1);
```

After this `sb` will be equal to `/` , this length check will have to change to `if (sb.length() > 1) {`.